pkg/operator/ceph/cluster/cluster.go

```diff
@@ -607,22 +607,31 @@ func (c *cluster) configureMsgr2() error {
 	}
 	monStore := config.GetMonStore(c.context, c.ClusterInfo)
 
-	// If the user has disabled encryption, remove the encryption settings
-	if c.Spec.Network.Connections == nil || c.Spec.Network.Connections.Encryption == nil || !c.Spec.Network.Connections.Encryption.Enabled {
+	encryptionEnabled := c.Spec.Network.Connections != nil &&
+		c.Spec.Network.Connections.Encryption != nil &&
+		c.Spec.Network.Connections.Encryption.Enabled
+
+	if encryptionEnabled {
+		logger.Infof("setting msgr2 encryption mode to %q", encryptionSetting)
+		if err := monStore.SetAll("global", encryptionGlobalConfigSettings); err != nil {
+			return err
+		}
+	} else {
 		encryptionConfig := []config.Option{}
 		for k := range encryptionGlobalConfigSettings {
 			encryptionConfig = append(encryptionConfig, config.Option{Who: "global", Option: k})
 		}
 		if err := monStore.DeleteAll(encryptionConfig...); err != nil {
 			return errors.Wrap(err, "failed to delete msgr2 encryption settings")
 		}
-	} else {
-		logger.Infof("setting msgr2 encryption mode to %q", encryptionSetting)
-		if err := monStore.SetAll("global", encryptionGlobalConfigSettings); err != nil {
-			return err
+
+		// set default rbd map options if msgr2 is enabled and encryption is disabled
+		if c.Spec.Network.Connections != nil && c.Spec.Network.Connections.RequireMsgr2 {
+			if err := monStore.SetAll("global", map[string]string{"rbd_default_map_options": defaultRBDMapOptions}); err != nil {
```

Comment:
This is wrong for the "encryption is disabled" case since `defaultRBDMapOptions` sets `ms_mode` to `secure`.  In this case, `ms_mode` should be set to `prefer-crc`.